### ISSUE TITLE
$this looses its context depending on where a file input is placed in the component view

### ISSUE THREAD

**Issue by tanthammar** [livewire/livewire#1401]
livewire v1.3.2
 `$this` looses its context depending on where the input is positioned in the form. 
If the field is in the top of the form, $this = the livewire component.
If it is positioned at the end of the form `$this` becomes the LivewireViewCompilerEngine.

Example:
```blade
@error('photos.*')
   <pre>@json($photos)</pre>
   @php $this->deleteAllTempFiles('photos') @endphp
@enderror 
```
Error:
 `Call to undefined method Livewire\LivewireViewCompilerEngine::deleteAllTempFiles()`.

Here you can see the @json($this) in the middle of the form = the livewire component

![](https://cdn.discordapp.com/attachments/698231501501890600/746480557092896858/unknown.png)

When the field is in the end of the form it becomes the LivewireViewCompilerEngine.
![](https://cdn.discordapp.com/attachments/698231501501890600/746481056655474718/unknown.png)

Please note that it is an extremely bad idea :) to call a method, that sets a component property, if the form has errors. It causes a loop with ajax requests. (But that is not the topic of this question.)

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by tanthammar** [comment]
Solved with  #2786. THANK YOU @joshhanley
